pkg/collector/container.go

```diff
@@ -101,23 +106,32 @@ func Container(cfg config.Config, logger logr.Logger, agent v1alpha1.AmazonCloud
 		Env:             envVars,
 		EnvFrom:         agent.Spec.EnvFrom,
 		Resources:       agent.Spec.Resources,
-		Ports:           portMapToList(ports),
+		Ports:           portMapToContainerPortList(ports),
 	}
 }
 
 func getContainerPorts(logger logr.Logger, cfg string) map[string]corev1.ContainerPort {
 	ports := map[string]corev1.ContainerPort{}
-	for _, p := range CloudwatchAgentPorts {
+	var servicePorts []corev1.ServicePort
+	configJSON, err := adapters.ConfigStructFromJSONString(cfg)
+	if err != nil {
+		logger.Error(err, "error parsing cw agent config")
+		servicePorts = PortMapToServicePortList(AppSignalsPortToServicePortMap)
+	} else {
+		servicePorts = getServicePortsFromCWAgentConfig(logger, configJSON)
+	}
+
+	for _, p := range servicePorts {
 		truncName := naming.Truncate(p.Name, maxPortLen)
 		if p.Name != truncName {
 			logger.Info("truncating container port name",
```

Comment:
np: can this be made in to debug instead of info ?

Reply:
This logger of type logr.logger doesn't have debug function. Only Error and Info are supported